### PR TITLE
Added database table and endpoint. Fixed main carousel to conditionally render by number of entries.

### PR DESCRIPTION
**Pull request by rtsuzaki**
- Replaced hardcoded data for image carousel with real data. (from new images database table).

- Fixed main carousel prev/next slide buttons to go to correct place in carousel. Made prev/next slide buttons to only appear if > 3 entries in carousel.

- Made main carousel only have one row if there are less than 4 room entries.

### CODE REVIEW

File: client/src/components/PriceSlider.jsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import Slider from 'rc-slider';
 import PropTypes from 'prop-types';
 
+
```
Comment (clfhhc) [inline client/src/components/PriceSlider.jsx]: style guide: no multiple empty lines

File: client/src/components/RoomListCarousel.jsx [hunk 2]
```diff
@@ -4,30 +4,55 @@ import RoomListEntry from './RoomListEntry';
...
   const firstHalf = props.displayedRooms.slice(0, halfLength);
   const secondHalf = props.displayedRooms.slice(halfLength, props.displayedRooms.length);
+
+  const roomImages = {};
+  props.displayedRooms.forEach((room) => {
+    roomImages[room.id] = [];
+  });
+
+  props.allImages.forEach((image) => {
+    if (roomImages.hasOwnProperty(image.roomId)) {
```
Comment (clfhhc) [inline client/src/components/RoomListCarousel.jsx]: airbnb eslint suggest not to use hasOwnProperty, use roomImages[image.roomId] truthiness instead

File: client/src/components/RoomListCarousel.jsx [hunk 3]
```diff
@@ -4,30 +4,55 @@ import RoomListEntry from './RoomListEntry';
...
       />
 
-      <div id="carousel">
-        <div id="room-carousel">
-          {firstHalf.map((room, index) => <RoomListEntry key={index} room={room} activeIndex={props.activeIndex} />)}
-        </div>
+      {displayTwoRows ? (
+        <div id="carousel">
+          <div id="room-carousel">
+            {firstHalf.map((room, index) => <RoomListEntry key={index} room={room} activeIndex={props.activeIndex} roomImages={roomImages} />)}
```
Comment (clfhhc) [inline client/src/components/RoomListCarousel.jsx]: if possible, use room id or room name as unique keys

File: client/src/components/RoomListCarouselArrow.jsx [hunk 4]
```diff
@@ -2,8 +2,11 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 function RoomListCarouselArrow(props) {
+  if (props.numberOfEntriesInCarousel <= 3) {
+    return null;
+  }
   return (
-    <div className="inline-block" onClick={props.clickFunction} >
+    <div onClick={()=>props.clickFunction(props.numberOfEntriesInCarousel)} className="flex-center">
       <svg className={`slide-arrow ${props.direction}`} viewBox="0 0 18 18" role="img">
```
Comment (clfhhc) [inline client/src/components/RoomListCarouselArrow.jsx]: better put svg viewBox and path under one single helper/unit

File: client/src/css/style.css [hunk 5]
```diff
@@ -59,55 +59,51 @@ body {
...
+  align-items: flex-start;
+  align-content: center;
+
 }
 
 #carousel {
   display:inline-block;
   box-sizing: border-box;
-  width:95%;
+  width:90%;
```
Comment (clfhhc) [inline client/src/css/style.css]: can set the buttons on both size "flex: 1 auto;" and then leave out the width property in here, and then in the parent div use "justify-content: stretch or  space-between"

File: database/index.js [hunk 6]
```diff
@@ -14,19 +14,32 @@ connection.connect((err) => {
...
 
-const insertRecords = (data) => {
-  connection.query('INSERT INTO roomlist (roomname, price, numberOfBedrooms, rating, numberOfReviews, urlToImage, roomType, instantBook) VALUES ?', [data]);
+const insertRoomlistRecords = (data) => {
+  connection.query('INSERT INTO roomlist (roomname, price, numberOfBedrooms, rating, numberOfReviews, roomType, instantBook) VALUES ?', [data]);
 };
 
-const getAllRecords = (sendCallback) => {
+const insertImagesRecords = (data) => {
+  connection.query('INSERT INTO images (urlToImage, roomId) VALUES ?', [data]);
```
Comment (clfhhc) [inline database/index.js]: can return this line to make it a promise and you can attach a callback/thenable later if you want

File: client/src/components/App.jsx [hunk 7]
```diff
@@ -118,23 +156,34 @@ class App extends React.Component {
...
-  //     .then((response) => {
-  //       this.setState({allRooms: response.data})
-  //     })
-  //     .catch((err) => {
-  //       throw err;
-  //     });
-  // }
+  setDisplayedRooms(rooms) {
+    if (rooms.length > 24) {
+      this.setState({ displayedRooms: rooms.slice(0, 25) });
```
[the remark this answers is not in the document]
Comment (rtsuzaki) [reply client/src/components/App.jsx]: oh yeah good point, I forgot slice would still work even if array is smaller. Thanks

**Pull request merged**